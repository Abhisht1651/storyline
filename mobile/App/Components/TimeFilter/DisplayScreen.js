// @flow
import React from 'react'
import { View } from 'react-native'
import styles from '../Styles/TimeFilterStyles/DisplayScreenStyle'

import TimeFilter from './TimeFilter'

const DisplayScreen = () => {
  const captureYears = (arr: Array<string>) => {
    console.log(arr)
  }

  const captureMonths = (arr: Array<string>) => {
    console.log(arr)
  }

  const btnPressHandler = () => {}
  const onCloseHandler = () => {}
  return (
    <View style={styles.container}>
      <TimeFilter
        getSelectedYears={captureYears}
        getSelectedMonths={captureMonths}
        onBtnPress={btnPressHandler}
        onClose={onCloseHandler}
      />
    </View>
  )
}

export default DisplayScreen
