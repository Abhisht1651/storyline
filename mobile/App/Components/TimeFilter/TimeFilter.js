// @flow

import React, { useEffect } from 'react'
import { View, Text, TouchableOpacity, Image, FlatList, Modal } from 'react-native'
import styles from '../Styles/TimeFilterStyles/TimeFilterStyle'
import Images from '../../Themes/Images'
import useHooks from './TimeFilterHooks'

type Props = {
  getSelectedYears: (arg: Array<string>) => void,
  getSelectedMonths: (arg: Array<string>) => void,
  onBtnPress: () => void,
  onClose: () => void,
  isModalVisible: boolean,
  headerTitle: string
}

const TimeFilter = (props: Props) => {
  const {
    getSelectedYears,
    getSelectedMonths,
    onBtnPress,
    onClose,
    isModalVisible,
    headerTitle
  } = props
  const {
    checked2020,
    checked2019,
    checked2018,
    checked2017,
    checked2016,
    checked2015,
    checkedJan,
    checkedFeb,
    checkedMar,
    checkedApr,
    checkedMay,
    checkedJun,
    checkedJul,
    checkedAug,
    checkedSept,
    checkedOct,
    checkedNov,
    checkedDec,
    clickHandler1,
    clickHandler2
  } = useHooks()
  const years = [
    { id: 2020, stateProp: checked2020 },
    { id: 2019, stateProp: checked2019 },
    { id: 2018, stateProp: checked2018 },
    { id: 2017, stateProp: checked2017 },
    { id: 2016, stateProp: checked2016 },
    { id: 2015, stateProp: checked2015 }
  ]

  const months1 = [
    { id: 'Jan', stateProp: checkedJan },
    { id: 'Feb', stateProp: checkedFeb },
    { id: 'Mar', stateProp: checkedMar },
    { id: 'Apr', stateProp: checkedApr },
    { id: 'May', stateProp: checkedMay },
    { id: 'Jun', stateProp: checkedJun }
  ]

  const months2 = [
    { id: 'Jul', stateProp: checkedJul },
    { id: 'Aug', stateProp: checkedAug },
    { id: 'Sept', stateProp: checkedSept },
    { id: 'Oct', stateProp: checkedOct },
    { id: 'Nov', stateProp: checkedNov },
    { id: 'Dec', stateProp: checkedDec }
  ]

  const selectedYears = []
  const selectedMonths = []
  useEffect(() => {
    if (checked2020) selectedYears.push('2020')
    if (checked2019) selectedYears.push('2019')
    if (checked2018) selectedYears.push('2018')
    if (checked2017) selectedYears.push('2017')
    if (checked2016) selectedYears.push('2016')
    if (checked2015) selectedYears.push('2015')
    getSelectedYears(selectedYears)
  }, [checked2020, checked2019, checked2018, checked2017, checked2016, checked2015])

  useEffect(() => {
    if (checkedJan) selectedMonths.push('Jan')
    if (checkedFeb) selectedMonths.push('Feb')
    if (checkedMar) selectedMonths.push('Mar')
    if (checkedApr) selectedMonths.push('Apr')
    if (checkedMay) selectedMonths.push('May')
    if (checkedJun) selectedMonths.push('Jun')
    if (checkedJul) selectedMonths.push('Jul')
    if (checkedAug) selectedMonths.push('Aug')
    if (checkedSept) selectedMonths.push('Sept')
    if (checkedOct) selectedMonths.push('Oct')
    if (checkedNov) selectedMonths.push('Nov')
    if (checkedDec) selectedMonths.push('Dec')
    getSelectedMonths(selectedMonths)
  }, [
    checkedJan,
    checkedFeb,
    checkedMar,
    checkedApr,
    checkedMay,
    checkedJun,
    checkedJul,
    checkedAug,
    checkedSept,
    checkedOct,
    checkedNov,
    checkedDec
  ])

  return (
    <Modal visible={isModalVisible} transparent>
      <View style={styles.modalStyle}>
        <View style={styles.container}>
          <View style={styles.view1}>
            <Text style={styles.headerText}>{headerTitle}</Text>
            <TouchableOpacity style={styles.closeIconContainer} activeOpacity={1} onPress={onClose}>
              <Image source={Images.closeIcon} style={styles.closeIcon} />
            </TouchableOpacity>
          </View>
          <View style={styles.view2}>
            <View style={styles.year}>
              <Text style={styles.view2HeaderText}>Year</Text>
              <FlatList
                contentContainerStyle={styles.flatList}
                data={years}
                renderItem={({ item }) => (
                  <View style={styles.listItem}>
                    <TouchableOpacity style={styles.circle} onPress={() => clickHandler1(item.id)}>
                      {item.stateProp && (
                        <View style={styles.checkedCircle}>
                          <Image source={Images.checkmark} />
                        </View>
                      )}
                    </TouchableOpacity>
                    <Text style={styles.listText}>{item.id}</Text>
                  </View>
                )}
                keyExtractor={(item) => item.id.toString()}
              />
            </View>

            <View style={styles.month}>
              <Text style={styles.view2HeaderText}>Month</Text>
              <View style={styles.monthContainer}>
                <View>
                  <FlatList
                    contentContainerStyle={styles.flatList}
                    data={months1}
                    renderItem={({ item }) => (
                      <View style={styles.listItem}>
                        <TouchableOpacity
                          style={styles.circle}
                          onPress={() => clickHandler2(item.id)}>
                          {item.stateProp && (
                            <View style={styles.checkedCircle}>
                              <Image source={Images.checkmark} />
                            </View>
                          )}
                        </TouchableOpacity>
                        <Text style={styles.listText}>{item.id}</Text>
                      </View>
                    )}
                    keyExtractor={(item) => item.id.toString()}
                  />
                </View>

                <View>
                  <FlatList
                    contentContainerStyle={styles.flatList}
                    data={months2}
                    renderItem={({ item }) => (
                      <View style={styles.listItem}>
                        <TouchableOpacity
                          style={styles.circle}
                          onPress={() => clickHandler2(item.id)}>
                          {item.stateProp && (
                            <View style={styles.checkedCircle}>
                              <Image source={Images.checkmark} />
                            </View>
                          )}
                        </TouchableOpacity>
                        <Text style={styles.listText}>{item.id}</Text>
                      </View>
                    )}
                    keyExtractor={(item) => item.id.toString()}
                  />
                </View>
              </View>
            </View>
          </View>
          <View style={styles.view3}>
            <TouchableOpacity style={styles.button} activeOpacity={1} onPress={onBtnPress}>
              <Text style={styles.btnText}>Apply</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </Modal>
  )
}
TimeFilter.defaultProps = {
  isModalVisible: true,
  headerTitle: 'Activity history'
}
export default TimeFilter
