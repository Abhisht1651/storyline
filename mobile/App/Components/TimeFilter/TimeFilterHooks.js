import { useState } from 'react'

export default function useHooks() {
  const [checked2020, setchecked2020] = useState(false)
  const [checked2019, setchecked2019] = useState(false)
  const [checked2018, setchecked2018] = useState(false)
  const [checked2017, setchecked2017] = useState(false)
  const [checked2016, setchecked2016] = useState(false)
  const [checked2015, setchecked2015] = useState(false)
  const [checkedJan, setcheckedJan] = useState(false)
  const [checkedFeb, setcheckedFeb] = useState(false)
  const [checkedMar, setcheckedMar] = useState(false)
  const [checkedApr, setcheckedApr] = useState(false)
  const [checkedMay, setcheckedMay] = useState(false)
  const [checkedJun, setcheckedJun] = useState(false)
  const [checkedJul, setcheckedJul] = useState(false)
  const [checkedAug, setcheckedAug] = useState(false)
  const [checkedSept, setcheckedSept] = useState(false)
  const [checkedOct, setcheckedOct] = useState(false)
  const [checkedNov, setcheckedNov] = useState(false)
  const [checkedDec, setcheckedDec] = useState(false)

  const clickHandler1 = (arg) => {
    switch (arg) {
      case 2020:
        setchecked2020(!checked2020)
        break
      case 2019:
        setchecked2019(!checked2019)
        break
      case 2018:
        setchecked2018(!checked2018)
        break
      case 2017:
        setchecked2017(!checked2017)
        break
      case 2016:
        setchecked2016(!checked2016)
        break
      case 2015:
        setchecked2015(!checked2015)
        break
      default:
        break
    }
  }

  const clickHandler2 = (arg) => {
    switch (arg) {
      case 'Jan':
        setcheckedJan(!checkedJan)
        break
      case 'Feb':
        setcheckedFeb(!checkedFeb)
        break
      case 'Mar':
        setcheckedMar(!checkedMar)
        break
      case 'Apr':
        setcheckedApr(!checkedApr)
        break
      case 'May':
        setcheckedMay(!checkedMay)
        break
      case 'Jun':
        setcheckedJun(!checkedJun)
        break
      case 'Jul':
        setcheckedJul(!checkedJul)
        break
      case 'Aug':
        setcheckedAug(!checkedAug)
        break
      case 'Sept':
        setcheckedSept(!checkedSept)
        break
      case 'Oct':
        setcheckedOct(!checkedOct)
        break
      case 'Nov':
        setcheckedNov(!checkedNov)
        break
      case 'Dec':
        setcheckedDec(!checkedDec)
        break

      default:
        break
    }
  }
  return {
    checked2020,
    checked2019,
    checked2018,
    checked2017,
    checked2016,
    checked2015,
    checkedJan,
    checkedFeb,
    checkedMar,
    checkedApr,
    checkedMay,
    checkedJun,
    checkedJul,
    checkedAug,
    checkedSept,
    checkedOct,
    checkedNov,
    checkedDec,
    clickHandler1,
    clickHandler2
  }
}
