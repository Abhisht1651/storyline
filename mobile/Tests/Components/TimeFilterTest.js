import { shallow } from 'enzyme'
import React from 'react'
import { renderHook, act } from '@testing-library/react-hooks'
import useHooks from '../../App/Components/TimeFilter/TimeFilterHooks'

import TimeFilter from '../../App/Components/TimeFilter/TimeFilter'

describe('TimeFilter', () => {
  it('should match to snapshot', () => {
    const component = shallow(<TimeFilter />)
    expect(component).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<TimeFilter />)
    expect(wrapper.find('Text')).toHaveLength(4)
  })

  test('should change vale of checkedJan', () => {
    const { result } = renderHook(() => useHooks())
    const val = result.current.checkedJan
    act(() => {
      result.current.clickHandler2('Jan')
    })

    expect(result.current.checkedJan).toBe(!val)
  })
})
