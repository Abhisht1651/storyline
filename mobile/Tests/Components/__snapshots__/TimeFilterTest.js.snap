// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeFilter should match to snapshot 1`] = `
<Component
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2222234D",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#E6ECED",
          "borderTopLeftRadius": 15,
          "borderTopRightRadius": 15,
          "borderWidth": 1,
          "height": 805,
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#E6ECED",
            "flexDirection": "row",
            "height": 101.85714285714285,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Medium",
              "fontSize": 27,
              "letterSpacing": -0.12,
              "marginLeft": "auto",
              "textAlign": "center",
              "width": 260,
            }
          }
        >
          Activity history
        </Text>
        <ForwardRef
          activeOpacity={1}
          style={
            Object {
              "marginLeft": 167.5,
              "marginRight": 51.94,
            }
          }
        >
          <Image
            style={
              Object {
                "aspectRatio": 1,
                "height": 17.28,
              }
            }
          />
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "height": 423.85714285714283,
              "justifyContent": "space-between",
              "marginLeft": 50,
              "marginRight": 112,
              "marginTop": 49.285714285714285,
              "width": 138,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 27,
                "letterSpacing": -0.12,
                "marginBottom": 23,
              }
            }
          >
            Year
          </Text>
          <FlatList
            contentContainerStyle={
              Object {
                "flex": 1,
                "justifyContent": "space-around",
              }
            }
            data={
              Array [
                Object {
                  "id": 2020,
                  "stateProp": false,
                },
                Object {
                  "id": 2019,
                  "stateProp": false,
                },
                Object {
                  "id": 2018,
                  "stateProp": false,
                },
                Object {
                  "id": 2017,
                  "stateProp": false,
                },
                Object {
                  "id": 2016,
                  "stateProp": false,
                },
                Object {
                  "id": 2015,
                  "stateProp": false,
                },
              ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </View>
        <View
          style={
            Object {
              "height": 391,
              "justifyContent": "space-between",
              "marginRight": 122,
              "marginTop": 49.285714285714285,
              "width": 328,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1F1F60",
                "fontFamily": "BananaGrotesk-Medium",
                "fontSize": 27,
                "letterSpacing": -0.12,
                "marginBottom": 23,
              }
            }
          >
            Month
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 328.57142857142856,
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <View>
              <FlatList
                contentContainerStyle={
                  Object {
                    "flex": 1,
                    "justifyContent": "space-around",
                  }
                }
                data={
                  Array [
                    Object {
                      "id": "Jan",
                      "stateProp": false,
                    },
                    Object {
                      "id": "Feb",
                      "stateProp": false,
                    },
                    Object {
                      "id": "Mar",
                      "stateProp": false,
                    },
                    Object {
                      "id": "Apr",
                      "stateProp": false,
                    },
                    Object {
                      "id": "May",
                      "stateProp": false,
                    },
                    Object {
                      "id": "Jun",
                      "stateProp": false,
                    },
                  ]
                }
                disableVirtualization={false}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onEndReachedThreshold={2}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                updateCellsBatchingPeriod={50}
                windowSize={21}
              />
            </View>
            <View>
              <FlatList
                contentContainerStyle={
                  Object {
                    "flex": 1,
                    "justifyContent": "space-around",
                  }
                }
                data={
                  Array [
                    Object {
                      "id": "Jul",
                      "stateProp": false,
                    },
                    Object {
                      "id": "Aug",
                      "stateProp": false,
                    },
                    Object {
                      "id": "Sept",
                      "stateProp": false,
                    },
                    Object {
                      "id": "Oct",
                      "stateProp": false,
                    },
                    Object {
                      "id": "Nov",
                      "stateProp": false,
                    },
                    Object {
                      "id": "Dec",
                      "stateProp": false,
                    },
                  ]
                }
                disableVirtualization={false}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onEndReachedThreshold={2}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                updateCellsBatchingPeriod={50}
                windowSize={21}
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 228.35714285714286,
            "width": "100%",
          }
        }
      >
        <ForwardRef
          activeOpacity={1}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#7B46E4",
              "borderRadius": 38.5,
              "height": 82.14285714285714,
              "justifyContent": "center",
              "marginTop": 94.46428571428571,
              "width": 533.9285714285714,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "BananaGrotesk-Bold",
                "fontSize": 24,
                "letterSpacing": -0.09,
              }
            }
          >
            Apply
          </Text>
        </ForwardRef>
      </View>
    </View>
  </View>
</Component>
`;
